samples/switchless/README.md

```diff
@@ -0,0 +1,184 @@
+worker thread never leave their respective execution contexts during the perceived function call.
+
+The calling thread and the worker thread need to exchange information twice during the call. When the switchless
+call is initiated, the caller needs to pass the `job` (encapsulating information regarding the function call in a
+ single object, for details see the next section) to the worker thread. And when the call finishes, the worker
+thread needs to pass the result back to the caller. Both exchanges need to be synchronized.
+
+While switchless calls save transition time, they require at least one additional thread to service the calls.
+More threads typically means more competition of the CPU cores and more context switches, hurting the performance.
+Whether to make a particular function switchless has to weigh the associated costs and savings. In general, the
+good candidates for switchless calls are functions that are: 1) short, thus the transition takes relatively high
+percentage of the overall execution time of the call; and 2) called frequently, so the savings in transition time
+add up.
+
+## How does OpenEnclave support switchless OCALLs
+
+OpenEnclave only supports synchronous switchless OCALLs currently. When the caller within an enclave makes a
+switchless OCALL, the trusted OpenEnclave runtime creates a `job` out of the function call. The `job` object
+includes information such as the function ID, the parameters marshaled into a buffer, and a buffer for holding the
+return value(s). The job is posted to a shared memory region which both the enclave and the host can access.
+
+A host worker thread checks and retrieves `job` from the shared memory region. It uses the untrusted OpenEnclave
+runtime to process the `job` by unmarshaling the parameters, then dispatching to the callee function, and finally
+relaying the result back to the trusted OpenEnclave runtime, which is further forwarded back to the caller.
+
+To support simultaneous switchless OCALLs made from enclaves, the host workers are multi-threaded. OpenEnclave
+allows users to configure how many host worker threads are to be created for servicing switchless OCALLs. The
+following example illustrates how to do that. A word of caution is that too many host worker threads might increase
+competition of cores between threads and degrade the performance. Therefore, if a enclave has switchless calls
+enabled, OpenEnclave caps the number of host worker threads for it to the number of enclave threads specified.
+
+With the current implementation, we recommend users to avoid using more host worker threads than the minimum of:
+
+1. the number of simultaneously active enclave threads, and
+2. the number of cores that are potentially available to host worker threads.
+
+For example, on a 4-core machine, if the number of the simultaneously active enclave threads is 2, and there is no
```

Comment:
grammar: change "is no" to "are no"

Reply:
Would still like this nit fixed, but not a blocking issue.

Reply:
Changes two occurrences.